main/src/library/List.flix

```diff
@@ -149,6 +149,40 @@ namespace List {
         case x :: rs => if (x == a) true else memberOf(a, rs)
     }
 
+    ///
+    /// Optionally finds the smallest element of `xs` according to the `Order` instance of `a`.
+    ///
+    pub def minimum[a : Order](xs: List[a]): Option[a] = minimumBy((x, y) -> x <=> y, xs)
+
+    ///
+    /// Optionally finds the smallest element of `xs` according to the given comparator `cmp`.
+    ///
+    pub def minimumBy(cmp: (a, a) -> Comparison, xs: List[a]): Option[a] = match xs {
+        case Nil => None
+        case y :: ys => Some(foldLeft((acc, z) ->
```

Comment:
Suggestion:

Instead of `match cmp(acc, z)` just do `Order.min(acc, z)`.

(This applies everywhere).

Reply:
In fact, I think you can rewrite most of these minimumBy and maximumBy to simply:

```
pub def minimumBy(cmp: (a, a) -> Comparison, xs: List[a]): Option[a] = reduce(Order.min, xs)
```

Right? :)

Reply:
Oh that's certainly elegant! But we don't use the given `cmp` function then?
It could be used for `minimum`, though.

Reply:
Maybe if we add these
```
minBy(cmp: (a, a,) -> Comparison): a
maxBy(cmp: (a, a,) -> Comparison): a
```
To `Order`? Then we can use `reduce(Order.minBy(cmp), xs)`?

Reply:
I've gone ahead and refactored it, let me know what you think :)